### PR TITLE
fix: order of volumesnapshot syncers

### PR DESCRIPTION
**Pull request by FabianKramm**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

Moved CRD creation into the mapper as the index for the mappers are called before the Init function for controllers was called. To avoid confusion the Initializer function was removed as well.

### CODE REVIEW

File: pkg/mappings/resources/volumesnapshotcontents.go [hunk 1]
```diff
@@ -28,3 +34,313 @@ func CreateVolumeSnapshotContentsMapper(ctx *synccontext.RegisterContext) (mappi
 		return vVSC.Annotations[constants.HostClusterVSCAnnotation]
 	})
 }
+
+var volumeSnapshotContentsCRD = `
```
Comment (facchettos) [inline pkg/mappings/resources/volumesnapshotcontents.go:38]: wouldn't that be a little nicer with embed?

File: pkg/patcher/patcher.go [hunk 2]
```diff
@@ -46,6 +46,8 @@ func NewSyncerPatcher(ctx *synccontext.SyncContext, pObj, vObj client.Object, op
 		return nil, fmt.Errorf("create virtual patcher: %w", err)
 	}
 
+	vPatcher.direction = "virtual"
+	pPatcher.direction = "host"
```
Comment (facchettos) [inline pkg/patcher/patcher.go:50]: is this going to be required for most patchers that need logging? I'd put it in the constructor if that's the case

Comment (FabianKramm) [reply pkg/patcher/patcher.go:50]: Sure let me put it in the constructor

File: pkg/util/crds.go [hunk 3]
```diff
@@ -17,6 +17,11 @@ import (
 )
 
 func EnsureCRD(ctx context.Context, config *rest.Config, manifest []byte, groupVersionKind schema.GroupVersionKind) error {
+	// TODO: find a better way to deal with this
+	if config.Host == "unit-test-client" {
+		return nil
+	}
+
```
Comment (facchettos) [inline pkg/util/crds.go:24]: could we have a `func` variable inside the util package that we can replace at runtime with the fake manager? 

something like 
```go
var ensureCrd func(ctx context.Context, config *rest.Config, manifest []byte, groupVersionKind schema.GroupVersionKind) error {
      create/update the crd
}

func UnitTestSetup(){
   ensureCRD = func(...)error{
    return nil
   }
}
```

Comment (FabianKramm) [reply pkg/util/crds.go:24]: Oh yeah good idea, let me do that

File: pkg/controllers/syncer/testing/manager.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ type fakeManager struct {
 
 func (f *fakeManager) SetFields(interface{}) error { return nil }
 
-func (f *fakeManager) GetConfig() *rest.Config { return &rest.Config{Host: "127.0.0.1"} }
+func (f *fakeManager) GetConfig() *rest.Config { return &rest.Config{Host: "unit-test-client"} }
```
Comment (facchettos) [inline pkg/controllers/syncer/testing/manager.go:33]: is it still used?

**Review by facchettos — APPROVED** [review]

File: pkg/controllers/syncer/testing/manager.go (hunk 4 above)
Comment (FabianKramm) [reply pkg/controllers/syncer/testing/manager.go:33]: I think this actually is still a good idea as this shows up in tests a lot nicer than 127.0.0.1

**Pull request merged**